infra/gravity/terraform.go

```diff
@@ -197,11 +197,18 @@ func makeDynamicParams(baseConfig ProvisionerConfig) (*cloudDynamicParams, error
 
 	param.homeDir = filepath.Join("/home", param.user)
 
+	// this seems like an awkward way to set a default value, there must be a better pattern
+	// -- 2020-02 wdella
+	if baseConfig.TerraformPluginDir == "" {
+		baseConfig.TerraformPluginDir = "/etc/terraform/plugins"
+	}
+
```

Comment:
I really don't like this location as a home for the default value.  I'd prefer that it were set closer to where `PluginDir` is declared in infra/terraform/config.go, and this just passed along a nil/empty string value.

It seems like there is a recommended pattern to do this here:

https://golang.org/doc/effective_go.html#composite_literals

I haven't quite got the hang of it yet though.  Is this the right way to go?

Reply:
I think this is fine. If you want to set defaults in .../config.go we can modify `Validate` to something like `ValidateAndSetDefaults` and set default values as we validate the config.

Reply:
> I think this is fine.

When you say "this" are you 1) approving of the current code, or 2) suggesting constructors with composite literals are the way to go?

Reply:
Removed hedgy comment in f861f38.